electron/main/buffer.js

```diff
@@ -1,23 +1,29 @@
-import fs from "fs"
 import { join } from "path"
 import { app } from "electron"
 import CONFIG from "../config"
 import { isDev } from "../detect-platform"
+import os from "node:os"
+import * as jetpack from "fs-jetpack"
 
+const untildify = (pathWithTilde) => {
+    const homeDirectory = os.homedir();
+    return homeDirectory
+      ? pathWithTilde.replace(/^~(?=$|\/|\\)/, homeDirectory)
+      : pathWithTilde;
+}
 
-export function getBufferFilePath() {
+export function getBufferFilePath(path) {
     let defaultPath = app.getPath("userData")
-    let configPath = CONFIG.get("settings.bufferPath")
-    let bufferPath = configPath.length ? configPath : defaultPath
-    let bufferFilePath = join(bufferPath, isDev ? "buffer-dev.txt" : "buffer.txt")
-    try {
-        // use realpathSync to resolve a potential symlink
-        return fs.realpathSync(bufferFilePath)
-    } catch (err) {
-        // realpathSync will fail if the file does not exist, but that doesn't matter since the file will be created
-        if (err.code !== "ENOENT") {
-            throw err
-        }
-        return bufferFilePath
+    let configPath = path ? path : CONFIG.get("settings.bufferPath")
+    // Expand the given path, preferring the config path to the default.
+    let bufferPath = untildify(configPath ? configPath : defaultPath)
+    let exists = jetpack.exists(bufferPath);
+    // Use the given path if it's an existing file or we will create a new one.
```

Comment:
This changes the behavior to allow a filename to be specified directly in the config. We could always derive a base dir from the path if there are additional metadata files in the future (as noted in https://github.com/heyman/heynote/pull/27#issuecomment-1868261276)

Reply:
We could do that, but by letting users specify a file I think it's reasonable of them to assume that Heynote isn't going to start pollute the directory of that file with other files (e.g. if they set the path to `~/Dropbox/heynote-buffer.txt`). But if they specify a directory it wouldn't be strange if we added more files there.

Reply:
Yeah good point, I think the `userData` dir is the right place for other settings and the only thing that needs to be configurable is the path to the buffer itself

Reply:
For most other settings, yes, `userData` is the way to go. But I imagine that we *might* want to add some kind of metadata that is tied to the buffer which *would* make sense to store together with the buffer file. Therefore I think it's just simpler if we only support specifying a directory.

Reply:
Makes sense. You could always store such metadata in a header, up to you

Reply:
fwiw with the current code if there is a full filepath specified in the config the app will crash with. The config could be sanity checked before loading, but would also only happen if a user manually changed the config instead of using the file dialog.

```
app threw an error during load
Error: ENOTDIR: not a directory, lstat '/Users/joshblum/Downloads/tes6.txt/buffer-dev.txt'
    at Object.realpathSync (node:fs:2655:7)
    at Object.realpathSync (node:electron/js2c/asar_bundle:2:5358)
    at getBufferFilePath (/Users/joshblum/code/heynote/dist-electron/main/index.js:33074:23)
    at loadBuffer (/Users/joshblum/code/heynote/dist-electron/main/index.js:33140:15)
    at Module.<anonymous> (/Users/joshblum/code/heynote/dist-electron/main/index.js:33368:1)
    at Module._compile (node:internal/modules/cjs/loader:1271:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1326:10)
    at Module.load (node:internal/modules/cjs/loader:1126:32)
    at Module._load (node:internal/modules/cjs/loader:967:12)
    at l._load (node:electron/js2c/asar_bundle:2:13642)
```